src/components/DashboardEditor/DashboardEditor.jsx

```diff
@@ -379,34 +389,79 @@ const DashboardEditor = ({
     setIsImageGalleryModalOpen(false);
   };
 
-  const commonCardProps = (cardConfig, isSelected) => ({
-    key: cardConfig.id,
-    tooltip: cardConfig.description,
-    availableActions: { clone: true, delete: true },
-    onCardAction: (id, actionId, payload) => {
-      if (actionId === CARD_ACTIONS.CLONE_CARD) {
-        onDuplicateCard(id);
-      } else if (actionId === CARD_ACTIONS.DELETE_CARD) {
-        onRemoveCard(id);
-      } else if (actionId === CARD_ACTIONS.ON_CARD_CHANGE) {
-        handleOnCardChange(update(cardConfig, payload));
-      }
-    },
-    tabIndex: 0,
-    onKeyDown: (e) => handleKeyDown(e, onSelectCard, cardConfig.id),
-    onClick: () => handleOnClick(onSelectCard, cardConfig.id),
-    className: `${baseClassName}--preview__card`,
-    isSelected,
-    // Add the show gallery to image card
-    onBrowseClick:
-      cardConfig.type === CARD_TYPES.IMAGE && isNil(cardConfig.content?.src)
-        ? handleShowImageGallery
-        : undefined,
-    validateUploadedImage:
-      cardConfig.type === CARD_TYPES.IMAGE
-        ? onValidateUploadedImage
-        : undefined,
-  });
+  const commonCardProps = useCallback(
```

Comment:
Not part of this PR, but I think we should change the name to `getCommonCardProps` or something to indicate that this is a function.